inc/classes/admin/settings/class-settings.php

```diff
@@ -543,4 +565,69 @@ function( $url ) {
 			)
 		);
 	}
+
+	/**
+	 * Sanitize a font file path.
+	 *
+	 * @since 3.6
+	 *
+	 * @param  string $file Filepath to sanitize.
+	 * @return string|bool  Sanitized filepath. False if not a font file.
+	 */
+	private function sanitize_font( $file ) {
+		if ( ! is_string( $file ) ) {
+			return false;
+		}
+
+		$file = trim( $file );
+
+		if ( empty( $file ) ) {
+			return false;
+		}
+
+		$file = wp_parse_url( $file );
+
+		if ( ! empty( $file['host'] ) && $this->get_host() !== $file['host'] ) {
+			return false;
+		}
+
+		$file = '/' . trim( $file['path'], '/' );
+
+		$ext     = strtolower( pathinfo( $file, PATHINFO_EXTENSION ) );
+		$formats = [
+			'eot',
+			'otf',
+			'ttf',
+			'svg',
+			'woff',
+			'woff2',
+		];
```

Comment:
Move this array to a property. Why? Avoids creating the array each time this callback is invoked, which results in a micro-optimization for execution time and memory space.

Reply:
Done in https://github.com/wp-media/wp-rocket/pull/2555/commits/ed57696b00a089b73997a90df3f396dab7e3b3e9.